### PR TITLE
Changes in StartRecordingCall Request models for Mixed Audio Only Feature

### PR DESCRIPTION
**Pull request by ninikasharma**
…udio Feature.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above. Here is the swagger spec PR - https://github.com/Azure/azure-rest-api-specs/pull/15618
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public CallingServerClientOptions(ServiceVersion version = LatestVersion)
         {
             ApiVersion = version switch
             {
-                ServiceVersion.V2021_06_15_Preview => "2021-06-15-preview",
```
Comment (arifsaikat-microsoft) [inline sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs:30]: Should we keep the older values in addition to the new values in the switch?

Comment (ninikasharma) [reply sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs:30]: That makes sense, Let me keep both. But "LatestVersion" could be just single version. That needs to be the latest.

File: sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs [hunk 2]
```diff
@@ -265,7 +277,7 @@ public virtual Response<StartCallRecordingResult> StartRecording(Uri recordingSt
         /// </summary>
         /// <param name="recordingId">The recording id to get the state of.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual async Task<Response<CallRecordingProperties>> GetRecordingStateAsync(string recordingId, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<CallRecordingProperties>> GetRecordingStateAsync(string recordingId,  CancellationToken cancellationToken = default)
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs:280]: nit: extra space

File: sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs [hunk 3]
```diff
@@ -241,7 +247,10 @@ public virtual async Task<Response<StartCallRecordingResult>> StartRecordingAsyn
         /// </summary>
         /// <param name="recordingStateCallbackUri">The uri to send state change callbacks.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual Response<StartCallRecordingResult> StartRecording(Uri recordingStateCallbackUri, CancellationToken cancellationToken = default)
+        /// <param name="recordingContentType">Contenttype for recording.</param>
+        /// <param name="recordingChannelType">Channeltype for recording.</param>
+        /// <param name="recordingFormatType">Channeltype for recording.</param>
+        public virtual Response<StartCallRecordingResult> StartRecording(Uri recordingStateCallbackUri, RecordingContentType? recordingContentType = null, RecordingChannelType? recordingChannelType = null, RecordingFormatType? recordingFormatType = null, CancellationToken cancellationToken = default)
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs:253]: RecordingContentType, RecordingChannelType and RecordingFormatType are currently in the Generated folder and will not be exposed in the SDK. To expose these models, create a class in the Models folder, similar to this class: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.PhoneNumbers/src/Models/PhoneNumberCost.cs and then regenerate the code from swagger.

**Review by minnieliu — COMMENTED** [review]
Please also consider updating the README file so that users understand how to use this new change in the SDK.

**Review by minnieliu — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs [hunk 4]
```diff
@@ -41,7 +42,13 @@ public enum ServiceVersion
             /// The Beta of the CallingServer service.
             /// </summary>
 #pragma warning disable CA1707 // Identifiers should not contain underscores
-            V2021_06_15_Preview = 1
+            V2021_06_15_Preview = 1,
+#pragma warning restore CA1707 // Identifiers should not contain underscores
+            /// <summary>
+            /// The Beta of the CallingServer service.
```
Comment (Mohit-Chakraborty) [inline sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs:48]: Better to have distinct comments for the 2 versions.
Something like 'The 2021_08_30 version of the CallingServer service.' And similarly for the one above.

File: sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs [hunk 5]
```diff
@@ -217,7 +217,10 @@ public virtual async Task<Response> RemoveParticipantAsync(string participantId,
         /// </summary>
         /// <param name="recordingStateCallbackUri">The uri to send state change callbacks.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual async Task<Response<StartCallRecordingResult>> StartRecordingAsync(Uri recordingStateCallbackUri, CancellationToken cancellationToken = default)
+        /// <param name="recordingContentType">Contenttype for recording.</param>
+        /// <param name="recordingChannelType">Channeltype for recording.</param>
+        /// <param name="recordingFormatType">Channeltype for recording.</param>
```
Comment (Mohit-Chakraborty) [inline sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs:222]: Format type for recording.

File: sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs [hunk 6]
```diff
@@ -241,7 +247,10 @@ public virtual async Task<Response<StartCallRecordingResult>> StartRecordingAsyn
         /// </summary>
         /// <param name="recordingStateCallbackUri">The uri to send state change callbacks.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual Response<StartCallRecordingResult> StartRecording(Uri recordingStateCallbackUri, CancellationToken cancellationToken = default)
+        /// <param name="recordingContentType">Contenttype for recording.</param>
+        /// <param name="recordingChannelType">Channeltype for recording.</param>
+        /// <param name="recordingFormatType">Channeltype for recording.</param>
```
Comment (Mohit-Chakraborty) [inline sdk/communication/Azure.Communication.CallingServer/src/ServerCall.cs:252]: Format type for recording.

File: sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs [hunk 7]
```diff
@@ -317,8 +373,8 @@ public partial class ServerCall
...
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Communication.CallingServer.StartCallRecordingResult>> StartRecordingAsync(System.Uri recordingStateCallbackUri, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.Communication.CallingServer.StartCallRecordingResult> StartRecording(System.Uri recordingStateCallbackUri, Azure.Communication.CallingServer.RecordingContentType? recordingContentType = default(Azure.Communication.CallingServer.RecordingContentType?), Azure.Communication.CallingServer.RecordingChannelType? recordingChannelType = default(Azure.Communication.CallingServer.RecordingChannelType?), Azure.Communication.CallingServer.RecordingFormatType? recordingFormatType = default(Azure.Communication.CallingServer.RecordingFormatType?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Mohit-Chakraborty) [inline sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:376]: Should we remove the existing APIs or mark them as `EditorBrowsableState.Never` and change their impl to forward to the new methods.

**Pull request merged**